src/Rules/NameUniqueForCities.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Rules;
+
+use Closure;
+use Illuminate\Contracts\Validation\ValidationRule;
+use Lightit\Backoffice\City\Domain\Models\City;
+
+class NameUniqueForCities implements ValidationRule
+{
+    public function validate(string $attribute, mixed $value, Closure $fail): void
+    {
+        /** @phpstan-ignore-next-line */
+        $city = request()->route('city');
+        /** @var int $cityId */
+        $cityId = $city->id ?? null;
+        $existingCity = City::where('name', $value)->first();
+
+        if ($cityId) {
+            $currentCity = City::query()->find($cityId);
+            if ($currentCity && $currentCity->name === $value) {
```

Comment:
This is simplier

```suggestion
            if ($currentCity?->name === $value) {
```

Same for `$existingCity->id`

Reply:
In this other PR, she changed the logic to be more clear
https://github.com/florenciajasinski/Travel-Agency/pull/6/files#r2166872256